validator/project_validator.go

```diff
@@ -585,6 +590,28 @@ func ensureReferentialIntegrity(project *model.Project, distroIDs []string, dist
 				}
 			}
 			buildVariantTasks[task.Name] = true
+
+			if task.IsGroup {
```

Comment:
If you look at where this is defined, the yaml /bson tags are "-" indicating that this is just something we ourselves populate to help us translate these tasks/variants later on, so this isn't really something you can use in validation.

A more useful map may be [taskId] - > taskGroupName (for all tasks in task groups). Then you can look up each buildVariantTask in the map directly (rather than looping through each task groups), and if it exists, you also have the task group name right there as the value. 

I'm also not sure what revTgNameTaskMap is for but given that we're removing the "isGroup" as it's own case it's likely not needed.

Reply:
`revTgNameTaskMap` maps tasks to task groups, so essentially it is `[taskId] - > taskGroupName`. It was supposed to be a reverse map of `tgNameTaskMap` because `tgNameTaskMap` maps task groups to a slice of its tasks - `taskGroupName -> [taskId1, taskId2]`.